docs/design/pattern_matching.md

```diff
@@ -126,9 +126,16 @@ A name binding pattern is a pattern.
 -   _binding-pattern_ ::= _identifier_ `:` _expression_
 -   _proper-pattern_ ::= _binding-pattern_
 
+If the binding pattern is enclosed by a `var` pattern, it is a _variable binding
+pattern_. Otherwise, it is a _value binding pattern_.
+
 The _identifier_ specifies the name of the _binding_. The type of the binding is
 specified by the _expression_. The scrutinee is implicitly converted to that
-type if necessary. The binding is then _bound_ to the converted value.
+type if necessary. If the pattern is a variable binding pattern, the scrutinee
+is then converted to an initializing expression, and used to initialize a
+complete object allocated by the `var` pattern, and the name is _bound_ to that
+object. Otherwise, the scrutinee is converted to a value expression, and the
+name is bound to that expression.
```

Comment:
I'd prefer that we just say that the scrutinee is used to initialize the object here, and leave the "converted to an initializing expression" part to be dealt with by the rules for initialization.